core/src/main/java/org/elasticsearch/action/bulk/BulkShardRequest.java

```diff
@@ -93,4 +91,9 @@ public void readFrom(StreamInput in) throws IOException {
         }
         refresh = in.readBoolean();
     }
+
+    @Override
+    public String toString() {
+        return "shard bulk {" + shardId() + "}";
```

Comment:
Same thing. As in `ShardFlushRequest`; I think the underlying request should be concatenated here.